python/dgl/convert.py

```diff
@@ -1643,8 +1643,107 @@ def _batcher(lst):
+                {key: F.squeeze(feat_dict[key], 0) for key in node_attrs}
+            )
+    if edge_attrs is not None:
+        for _, _, attr in nx_graph.edges(data=True):
+            eid = attr["id"]
+            feat_dict = g._get_e_repr(0, eid)
+            attr.update(
+                {key: F.squeeze(feat_dict[key], 0) for key in edge_attrs}
+            )
+    return nx_graph
+
+
+def _to_networkx_heterogeneous(
+    g, node_attrs, edge_attrs, ntype_attr, etype_attr
+):
+    nx_graph = nx.MultiDiGraph()
+
+    num_nodes_per_ntype = [g.num_nodes(ntype) for ntype in g.ntypes]
+    offset_per_ntype = np.insert(np.cumsum(num_nodes_per_ntype), 0, 0)
+    num_edges_per_etype = [g.num_edges(etype) for etype in g.canonical_etypes]
+    offset_per_etype = np.insert(np.cumsum(num_edges_per_etype), 0, 0)
+
+    for ntype in g.ntypes:
+        nodes = F.asnumpy(g.nodes(ntype))
+        node_offset = int(offset_per_ntype[g.get_ntype_id(ntype)])
+
+        nx_graph.add_nodes_from(nodes + node_offset, **{ntype_attr: ntype})
+
+    for etype in g.canonical_etypes:
+        srctype, _, dsttype = etype
+        src, dst = g.all_edges(etype=etype, order="eid")
+        src_offset = int(offset_per_ntype[g.get_ntype_id(srctype)])
+        dst_offset = int(offset_per_ntype[g.get_ntype_id(dsttype)])
+
+        src = F.asnumpy(src) + src_offset
+        dst = F.asnumpy(dst) + dst_offset
+
+        for eid, (u, v) in enumerate(zip(src, dst)):
+            eid_offset = eid + int(offset_per_etype[g.get_etype_id(etype)])
+            nx_graph.add_edge(u, v, id=eid_offset, **{etype_attr: etype})
+
+    if node_attrs is not None:
+        assert (
+            ntype_attr not in node_attrs
+        ), f"'{ntype_attr}' already used as node type attribute"
+        for nid, attr in nx_graph.nodes(data=True):
+            ntype_id = g.get_ntype_id(attr[ntype_attr])
+            n_offset = int(offset_per_ntype[ntype_id])
+
+            feat_dict = g._get_n_repr(ntype_id, nid - n_offset)
+            attr.update(
+                {
+                    key: F.squeeze(feat_dict[key], 0)
+                    for key in node_attrs
+                    if key in feat_dict
+                }
+            )
+
+    if edge_attrs is not None:
+        assert (
+            etype_attr not in edge_attrs
+        ), f"'{etype_attr}' already used as edge type attribute"
+        for _, _, attr in nx_graph.edges(data=True):
+            etype_id = g.get_etype_id(attr[etype_attr])
+            eid = attr["id"]
+            eid_offset = int(offset_per_etype[etype_id])
+
+            feat_dict = g._get_e_repr(etype_id, eid - eid_offset)
+            attr.update(
+                {
+                    key: F.squeeze(feat_dict[key], 0)
+                    for key in edge_attrs
+                    if key in feat_dict
+                }
+            )
+
+    return nx_graph
+
+
+def to_networkx(
+    g,
+    node_attrs=None,
+    edge_attrs=None,
+    ntype_attr="type",
```

Comment:
I suggest changing to "ntype" to be consistent with the terminology used by DGL.

Reply:
Sounds good. Done.